### PR TITLE
[#10299] Fix some issues from UAT round 3

### LINKED ISSUE #10302: Link recovery: submission link ends up in session results page
Tried to retrieve submission links. Received this:
image

Clicking on the link resulted in this:
image

Note that I've already joined the course and I'm also an instructor in that course, if that is relevant. I can access the submission page via my student home page.

### LINKED ISSUE #10303: Numerical scale questions: some valid values are not accepted
V7

Example:
image

Possibly due to floating point number problems. Might affect other places too.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Part of #10299, fixes #10302, fixes #10303 

Screen Shot 2020-07-13 at 10 09 53 PM

- Rubric question submission: V6 has different interface in mobile view

Screen Shot 2020-07-13 at 10 11 11 PM

- Instructor course edit: V6 has timezone offset information in the dropdown (
- Instructor course edit: for instructor access level, unless in "edit" mode, just show the existing access level name (i.e. no need to display all other access levels)
- Instructor search: Not working if not both "students" and "questions, responses, comments on responses" are selected.
- MSQ question: minimum selectable choice should be 1 instead of 2.
- Preview submission form as student shows the form in readonly mode. It shouldn’t be like that. Only the submission button should be disabled.

### CODE REVIEW

File: src/web/app/pages-session/session-submission-page/session-submission-page.component.html [hunk 1]
```diff
@@ -77,7 +77,7 @@ <h4>Previewing Session as
 <tm-question-submission-form *ngFor="let questionSubmissionForm of questionSubmissionForms; let i = index; trackBy: trackQuestionSubmissionFormByFn"
                              [attr.id]="questionSubmissionForm.feedbackQuestionId"
                              [(formModel)]="questionSubmissionForms[i]" [formMode]="getQuestionSubmissionFormMode(questionSubmissionForm)"
-                             [isDisabled]="isSubmissionFormsDisabled"
+                             [isDisabled]="isSubmissionFormsDisabled && !this.previewAsPerson"
```
Comment (xpdavid) [inline src/web/app/pages-session/session-submission-page/session-submission-page.component.html:80]: Shouldn't we just remove this line?

https://github.com/TEAMMATES/teammates/blob/5deb81a13108950736b7484f938f1f989f962ce2/src/web/app/pages-session/session-submission-page/session-submission-page.component.ts#L147-L150

File: src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html [hunk 2]
```diff
@@ -10,9 +10,26 @@
...
       </td>
     </tr>
   </tbody>
 </table>
+<div class="mobile-view">
+  <div class="card" *ngFor="let rubricDescriptionRow of questionDetails.rubricDescriptions; let i = index;">
+    <div class="card-header bg-light">
+      {{ questionDetails.rubricSubQuestions[i] }}
+    </div>
+    <div class="card-body"
```
Comment (xpdavid) [inline src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html:24]: We should have `(click)="selectAnswer(i, j)"` here?

Comment (wkurniawan07) [reply src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html:24]: No. `j` is not defined at this block.

File: src/web/app/pages-session/session-submission-page/session-submission-page.component.html (hunk 1 above)
Comment (wkurniawan07) [reply src/web/app/pages-session/session-submission-page/session-submission-page.component.html:80]: Removing that will cause the "Submit Feedback" button to be enabled

Comment (xpdavid) [reply src/web/app/pages-session/session-submission-page/session-submission-page.component.html:80]: Then I think we should update the `[disabled]` attribute in the submit button to be `xx || this.previewAsPerson`. The current looks weird as `isSubmissionFormsDisabled` is controlled by `previewAsPerson` while `isDisabled` also depends on `previewAsPerson`.

File: src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html (hunk 2 above)
Comment (xpdavid) [reply src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html:24]: The problem is that the model only get updated after `selectAnswer(i, j)` is called.

File: src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html [hunk 3]
```diff
@@ -10,9 +10,26 @@
...
+         [ngClass]="{ 'row-answered': i < responseDetails.answer.length && responseDetails.answer[i] !== RUBRIC_ANSWER_NOT_CHOSEN }">
+      <div *ngFor="let rubricDescriptionCell of rubricDescriptionRow; let j = index;">
+        <label>
+          <input type="radio" [disabled]="isDisabled"
```
Comment (xpdavid) [inline src/web/app/components/question-types/question-edit-answer-form/rubric-question-edit-answer-form.component.html:28]: We should add `(click)="selectAnswer(i, j)"` here.

File: src/web/app/pages-session/session-submission-page/session-submission-page.component.html (hunk 1 above)
Comment (wkurniawan07) [reply src/web/app/pages-session/session-submission-page/session-submission-page.component.html:80]: @xpdavid the submit button also depends on `isSubmissionFormsDisabled` so in this case no alternative is better than the other

**Pull request merged**